extensions/amp-analytics/linker-id-receiving.md

```diff
@@ -0,0 +1,55 @@
+### Linker ID Receiving
+#### Overview
+The purpose of this document is to illustrate in detail how the linker param is constructed, so that analytics vendors or publishers implementing their own soulutions are able to ingest this parameter on the destination page.
+
+#### Format
+The linker param will be formatted in the following structure:
+
+`<paramName>=<version>*<checkSum>*<idName1>*<idValue1>*<idName2>*<idValue2>...`
+
+The asterisk `*` will be used to seperate each piece of data.
+
+- `paramName` - Name of parameter given in configuration.
+- `version` - The version of the linker used to create the param.
+- `checkSum` - A unique value generated by the linker. You can find details about how this value is generated [here](#checksum)
+- `idName1*idValue1` This is where any key value pairs in the `ids` object will be included. The values will be "URL Safe" base64 encoded, while the keys will remain in their raw form.
+
+#### Example
+Take this example [linker configuration](./linker-id-forwarding):
+```javascript
+"linkers": {
+  "linker1" : {
+    "ids" : {
+      "cid": "CLIENT_ID(_ga)",
+      "uid" "QUERY_PARAM(uid)",
+    }
+    "proxyOnly" : false,
+    "enabled": true,
+  }
+}
+```
+
+The above config would generate a parameter similar to the following: `linker1=1*5epghs*cid*YW1wLVNsQjJnZmlpXzExZFdxdFNyM3pUNHc.*uid*MTIzNA..`.
+
+There are several parts to this url parameter. First you will see that the name of the param `linker1` matches the name from the configuration above. The param value contains several pieces of data seperated by the `*`delimiter.
+- The first value `1` indicates the version number of the linker used.
+- The second value `5epghs` is a [checksum](#checksum) generated by the linker.
+- After that you will see the two key value pairs given in the `ids` object. The linker has resolved any macros that may exist, base64 encoded them in a url-safe format, and then appended them to the parameter. This results in `cid*YW1wLVNsQjJnZmlpXzExZFdxdFNyM3pUNHc.`, and `uid*MTIzNA..`.
+
+#### Checksum
+The checksum is a unique value generated by the linker that can be used to prevent accidental Id sharing. Publishers and/or anayltics vendors can generate this checksum on the destination page and ensure that it matches. Since this checksum contains a time input, a common practice is to check it against a few values generated by adding or subtracting a time tolerance.
+
+It is generated from the following formula.
+
+`base36(CRC32(fingerprint*timestampRoundedInMin*key1*val1*key2*val2...))`
+
+- `fingerprint` - A semi-unique value for page visitors. Generated by using `User Agent + * + timezone + * + language`.
+[code](https://github.com/ampproject/amphtml/blob/d8b40f62bbfe217307a21e600633343d9938aa45/extensions/amp-analytics/0.1/linker.js#L67)
```

Comment:
code pointer with line num is not recommended, since it will be invalid after code change